api/src/main/java/org/pentaho/platform/api/genericfile/model/IGenericFile.java

```diff
@@ -79,13 +102,32 @@ default String getNameDecoded() {
    */
   String getParentPath();
 
+  /**
+   * Sets the path of the parent folder, as a string, if any.
+   * <p>
+   * The provider root folders do not have a parent.
+   * Otherwise, a valid generic file instance must have a non-null parent path.
+   *
+   * @param parentPath the path of the parent folder; may be null for provider root folders.
+   * @see #getPath()
+   */
+  void setParentPath( String parentPath );
+
   /**
    * Gets the type of generic file, one of: {@link #TYPE_FOLDER} or {@link #TYPE_FILE}.
    *
    * @see #isFolder()
    */
   String getType();
 
+  /**
+   * Sets the type of generic file, one of: {@link #TYPE_FOLDER} or {@link #TYPE_FILE}.
+   *
+   * @param type the type of generic file; must be one of: {@link #TYPE_FOLDER} or {@link #TYPE_FILE}.
+   * @see #isFolder()
+   */
+  void setType( String type );
```

Comment:
Ideally, should say that for `IGenericFolder`, `setType(.)` only accepts `TYPE_FOLDER`, and `getType()` always returns `TYPE_FOLDER`. Either here, or in the ``IGenericFolder` documentation.